### PR TITLE
EFK for 19.10

### PR DESCRIPTION
**Pull request by fatpat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ An Ansible role for elasticsearch. Specifically, the responsibilities of this ro
 | openio_elasticsearch_version              | string  | ES version                              |
 | openio_elasticsearch_sysctl               | dict    | Sysctl options to apply                 |
 | openio_elasticsearch_java_args            | string  | ES java args                            |
-| openio_elasticsearch_pid_directory        | string  | Path to ES pid directory                |
+| openio_elasticsearch_pid_dir              | string  | Path to ES pid directory                |
```
Comment (cdelgehier) [inline README.md]: no,  you can't change the wording
All oio roles defined `*pid_directory`

Please stay consistent

File: defaults/main.yml [hunk 2]
```diff
@@ -3,12 +3,19 @@
 openio_elasticsearch_namespace: "OPENIO"
 openio_elasticsearch_serviceid: "0"
 
-openio_elasticsearch_gridinit_dir: "/etc/gridinit.d/{{ openio_elasticsearch_namespace }}"
-openio_elasticsearch_gridinit_file_prefix: ""
+openio_elasticsearch_sysconfig_dir: "/etc/oio/sds/{{ openio_elasticsearch_namespace }}/{{ openio_elasticsearch_servicename }}"
```
Comment (cdelgehier) [inline defaults/main.yml]: please stay consistent with other roles
Define it in `vars` instead of `default`

File: defaults/main.yml [hunk 3]
```diff
@@ -3,12 +3,19 @@
...
+openio_elasticsearch_log_dir: "/var/log/oio/sds/{{ openio_elasticsearch_namespace }}/{{ openio_elasticsearch_servicename }}"
 
-openio_elasticsearch_bind_interface: "{{ ansible_default_ipv4.alias }}"
-openio_elasticsearch_bind_address:
-  "{{ hostvars[inventory_hostname]['ansible_' + openio_elasticsearch_bind_interface]['ipv4']['address'] }}"
+prometheus_bind_address: "{{ openio_bind_mgmt_address
+  | d(hostvars[inventory_hostname]['ansible_' + prometheus_bind_interface].ipv4.address) }}"
+
+openio_elasticsearch_bind_port: 6400
+openio_elasticsearch_bind_interface: "{{ openio_bind_mgmt_interface | d(ansible_default_ipv4.alias) }}"
```
Comment (cdelgehier) [inline defaults/main.yml]: either you choose openio_bind_mgmt_interface for the 3 roles, or openio_bind_interface but not a mix of both

File: defaults/main.yml [hunk 4]
```diff
@@ -40,13 +47,8 @@ openio_elasticsearch_java_args: "-Xms{{ openio_elasticsearch_memory }} -Xmx{{ op
   -Dlog4j.shutdownHookEnabled=false \
   -Dlog4j2.disable.jmx=true"
 
-openio_elasticsearch_pid_directory:
-  "/run/elasticsearch/{{ openio_elasticsearch_namespace }}/elasticsearch-{{ openio_elasticsearch_serviceid }}"
 openio_elasticsearch_memory: "1024M"
 
-openio_elasticsearch_http_port: 6400
-openio_elasticsearch_transport_tcp_port: 6450
```
Comment (cdelgehier) [inline defaults/main.yml:48]: no transport port ?

File: tasks/main.yml [hunk 5]
```diff
@@ -14,54 +14,40 @@
...
 - name: Ensure elasticsearch pid directory is persistant
   lineinfile:
     dest: /etc/tmpfiles.d/elasticsearch.conf
     create: true
-    line: "d {{ openio_elasticsearch_pid_directory }} 755 openio openio -"
+    line: "d {{ openio_elasticsearch_pid_dir }} 755 openio openio -"
     owner: openio
     group: openio
     mode: 0644
   tags: install
```
Comment (cdelgehier) [inline tasks/main.yml]: configure

File: tasks/main.yml [hunk 6]
```diff
@@ -102,12 +86,12 @@
 - name: Wait for Elasticsearch to be available
   wait_for:
     host: "{{ openio_elasticsearch_bind_address }}"
-    port: "{{ openio_elasticsearch_http_port }}"
+    port: "{{ openio_elasticsearch_bind_port }}"
     delay: 10
```
Comment (cdelgehier) [inline tasks/main.yml]: prefer `retry` instead of `delay`

**Review by cdelgehier — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (fatpat) [reply README.md]: fixed

File: defaults/main.yml (hunk 2 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

File: defaults/main.yml (hunk 4 above)
Comment (fatpat) [reply defaults/main.yml:48]: fixed

File: tasks/main.yml (hunk 6 above)
Comment (fatpat) [reply tasks/main.yml]: fixed

**Review by fatpat — COMMENTED** [review]
seess OK to me

File: tasks/main.yml (hunk 5 above)
Comment (fatpat) [reply tasks/main.yml]: fixed

File: defaults/main.yml (hunk 3 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

**Pull request merged**